Master-Game/World/BaseCollectible.cs

```diff
@@ -0,0 +1,24 @@
+﻿namespace MasterGame.World
+{
+    public class BaseCollectible
+    {
```

Comment:
if it's a collectible.. it should always be collectable. Else it would just be an entity or an Item? May be the base class needs to be ItemEntity instead of BaseCollectible. BaseCollectible needs to be a child of ItemEntity.